backend_new/test/integration/listing.e2e-spec.ts

```diff
@@ -216,4 +238,550 @@ describe('Listing Controller Tests', () => {
+      postmarkedApplicationsReceivedByDate: new Date(),
+      programRules: 'program rules',
+      rentalHistory: 'rental history',
+      requiredDocuments: 'required docs',
+      specialNotes: 'special notes',
+      waitlistCurrentSize: 0,
+      waitlistMaxSize: 100,
+      whatToExpect: 'what to expect',
+      status: ListingsStatusEnum.active,
+      displayWaitlistSize: true,
+      reservedCommunityDescription: 'reserved community description',
+      reservedCommunityMinAge: 66,
+      resultLink: 'result link',
+      isWaitlistOpen: true,
+      waitlistOpenSpots: 100,
+      customMapPin: false,
+      jurisdictions: {
+        id: jurisdictionA.id,
+      },
+      reservedCommunityTypes: {
+        id: reservedCommunityType.id,
+      },
+      listingFeatures: {
+        elevator: true,
+        wheelchairRamp: false,
+        serviceAnimalsAllowed: true,
+        accessibleParking: false,
+        parkingOnSite: true,
+        inUnitWasherDryer: false,
+        laundryInBuilding: true,
+        barrierFreeEntrance: false,
+        rollInShower: true,
+        grabBars: false,
+        heatingInUnit: true,
+        acInUnit: false,
+        hearing: true,
+        visual: false,
+        mobility: true,
+      },
+      listingUtilities: {
+        water: false,
+        gas: true,
+        trash: false,
+        sewer: true,
+        electricity: false,
+        cable: true,
+        phone: false,
+        internet: true,
+      },
+    };
+
+    const res = await request(app.getHttpServer())
+      .put(`/listings/${listing.id}`)
+      .send(val)
+      .expect(200);
+    expect(res.body.id).toEqual(listing.id);
+    expect(res.body.name).toEqual(val.name);
+  });
+
+  it('should create listing', async () => {
+    const jurisdictionA = await prisma.jurisdictions.create({
+      data: jurisdictionFactory(),
+    });
+    await unitTypeFactoryAll(prisma);
+    const unitType = await unitTypeFactorySingle(prisma, UnitTypeEnum.SRO);
+    const amiChart = await prisma.amiChart.create({
+      data: amiChartFactory(1, jurisdictionA.id),
+    });
+    const unitAccessibilityPriorityType =
+      await prisma.unitAccessibilityPriorityTypes.create({
+        data: unitAccessibilityPriorityTypeFactorySingle(),
+      });
+    const rentType = await prisma.unitRentTypes.create({
+      data: unitRentTypeFactory(),
+    });
+    const multiselectQuestion = await prisma.multiselectQuestions.create({
+      data: multiselectQuestionFactory(jurisdictionA.id),
+    });
+    const reservedCommunityType = await prisma.reservedCommunityTypes.create({
+      data: reservedCommunityTypeFactory(jurisdictionA.id),
+    });
+
+    const exampleAddress = {
+      city: 'Exygy',
+      state: 'CA',
+      zipCode: '94104',
+      street: '548 Market St',
+    };
+
+    const exampleAsset = {
+      fileId: randomUUID(),
+      label: 'example asset label',
+    };
+
+    const val: ListingPublishedCreate = {
```

Comment:
A lot of this val is very similar to the one in the above test (ListingPublishedEdit object). I wonder if there is a way to pull it out to make this test more readable?

Reply:
updated here and in the service spec! defo made those files a bit easier to parse ty ty